libcraft/core/src/dimension.rs

```diff
@@ -0,0 +1,82 @@
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(try_from = "String", into = "&'static str")]
+pub enum Dimension {
+    Overworld,
+    TheNether,
+    TheEnd,
+}
+
+impl Dimension {
+    pub fn dim_id(&self) -> i32 {
```

Comment:
To maintain consistency in libcore this should be renamed `id` and `from_id`, see for example https://github.com/feather-rs/feather/blob/main/libcraft/particles/src/particle.rs or https://github.com/feather-rs/feather/blob/main/libcraft/core/src/biome.rs.